project-api-tests/src/test/java/org/ballerina/projectapi/BalToolTest.java

```diff
@@ -0,0 +1,870 @@
+        envVariables.put(BALLERINA_DEV_CENTRAL, "true");
+        if (!isToolAvailableInCentral(toolId, tempWorkspaceDirectory, envVariables)) {
+            Assert.fail("Tool " + toolId + " is not available in central");
+        }
+    }
+
+    @BeforeGroups(value = "pull")
+    public void setupPullTests() {
+        // Remove the tool from bal-tools.toml if exists
+        BalToolsToml balToolsToml = BalToolsToml.from(toolEnvironments.get(ToolSubCommand.PULL).balToolsTomlPath);
+        BalToolsManifest balToolsManifest = BalToolsManifestBuilder.from(balToolsToml).build();
+        balToolsManifest.removeTool(toolId);
+        balToolsToml.modify(balToolsManifest);
+
+        // Delete the tool from local cache if exists
+        Path toolPath = toolEnvironments.get(ToolSubCommand.PULL).centralCachePath
+                .resolve(Path.of(orgName, packageName));
+        if (Files.exists(toolPath)) {
+            ProjectUtils.deleteDirectory(toolPath);
+        }
+    }
+
+    @Test(description = "Pull tool without specifying a version", groups = {"pull"})
+    public void testPullToolWithoutVersion() throws IOException, InterruptedException {
+        Process cmdExec = executeToolCommand(DISTRIBUTION_FILE_NAME, tempWorkspaceDirectory,
+                new ArrayList<>(Arrays.asList("pull", toolId)), toolEnvironments.get(ToolSubCommand.PULL).envVariables);
+        String cmdErrors = getString(cmdExec.getErrorStream());
+        if (!cmdErrors.isEmpty()) {
+            Assert.fail(OUTPUT_CONTAIN_ERRORS + cmdErrors);
+        }
+
+        // Validate the command output
+        String cmdOutput = getString(cmdExec.getInputStream());
+        String expectedOutput = readExpectedCmdOutsAsString("tool-pull-without-version.txt");
+        if (!cmdOutput.contains(expectedOutput)) {
+            Assert.fail(OUTPUT_NOT_CONTAINS_EXP_MSG + expectedOutput + "\nactual output:" + cmdOutput);
+        }
+
+        // check for tool availability in local env
+        BalToolsToml balToolsToml = BalToolsToml.from(toolEnvironments.get(ToolSubCommand.PULL).balToolsTomlPath);
+        Optional<BalToolsManifest.Tool> toolOpt = BalToolsManifestBuilder.from(balToolsToml).build()
+                .getActiveTool(toolId);
+        if (toolOpt.isEmpty()) {
+            Assert.fail("Tool " + toolId + " is not available in bal-tools.toml");
+        }
+        BalToolsManifest.Tool tool = toolOpt.get();
+        Assert.assertEquals(tool.id(), toolId);
+        Assert.assertEquals(tool.version(), latestVersion);
+        Assert.assertEquals(tool.org(), orgName);
+        Assert.assertEquals(tool.name(), packageName);
+
+        Path toolVersionCachePath = toolEnvironments.get(ToolSubCommand.PULL).centralCachePath
+                .resolve(Path.of(tool.org(), tool.name(), tool.version()));
+        if (!Files.exists(toolVersionCachePath)) {
+            Assert.fail("Tool " + toolId + ":" + latestVersion + " is not available in local cache");
+        }
```

Comment:
Let's add another assertion to verify the command: i.e. `bal <tool-cmd> --help`

Reply:
Applies other test cases as well.

Reply:
Fixed with 0b92fd97f16524ed01af439c88ff2b462485cd34